scripts/vint.py

```diff
@@ -0,0 +1,119 @@
+    config = dict()
+    args = [
+        "vint",
+        "--json",
+        "."
+    ]
+    compiler_output = list()
+    github_output = list()
+
+    # Execute the Vint command
+    def exec(self):
+        command = self.build_command()
+        print("-- COMMAND: ", command)
+
+        try:
+            process = subprocess.Popen(command, stdout = subprocess.PIPE, shell=True)
+            output = process.stdout.readlines()
+            process.wait()
+
+            return output
+
+        except subprocess.CalledProcessError as e:
+            print("Error in calling Vint; ", e.output)
+            return None
+        except:
+            print("Unexpected error:", sys.exc_info()[0])
+            return None
+
+    # Construct the command
+    def build_command(self):
+        return ' '.join(self.args)
+
+    # Main handler
+    def run(self):
+        self.compile()
+
+        for message in self.github_output:
+            print(message.replace('\n', '%0A').replace('\r', '%0D'))
+
+    # Compile the command and output
+    def compile(self):
+        output = self.exec()
+
+        if output == None:
+            print("No violations found.")
+            exit(0)
+
+        self.process_output(output)
+
+    # Process output to JSON only
+    def process_output(self, output):
+        for line in output:
+            try:
+                line = line.strip()
+                json_line = json.loads(line)
+
+                if "description" in json_line:
+                    # Accept this line
+                    self.compiler_output.append({"json": json_line})
+
+            # Not JSON, so skip it
+            except AttributeError:
+                print('Skipping line in output; ', line)
+            except ValueError:
+                print('Skipping line in output; ', line)
+
+    def generate_github_output(self):
+        for json_line in self.compiler_output:
+            gh_output = self.line_compiler_to_gh(json_line['json'])
+
+            if gh_output != None:
+                self.github_output.append(gh_output)
+
+    def line_compiler_to_gh(self, json_line):
+        # Validate that we have a file path
+        if 'file_path' not in json_line:
+            return None
+
+        level = json_line['severity']
+        message = f"{json_line['description']}.\nsee `{json_line['reference']}`"
+
+        if level == "style_problem":
+            return f"::notice file={json_line['file_path']},line={json_line['line_number']},col={json_line['column_number']}::{message}"
+
+        if level == "warning":
+            return f"::warning file={json_line['file_path']},line={json_line['line_number']},col={json_line['column_number']}::{message}"
+
+        if level == "error":
+            return f"::error file={json_line['file_path']},line={json_line['line_number']},col={json_line['column_number']}::{message}"
+
+        print("Line was missing information.")
+        return None
+
+    def __init__(self):
+        self.run()
```

Comment:
```suggestion
```
Unless this is a standard of some sort it's rather unexpected from an outsider perspective to see a constructor also implicitly run stuff.